src/OpenLoco/Paint/Paint.h

```diff
@@ -0,0 +1,158 @@
+    static_assert(sizeof(PaintStringStruct) == 0x20);
+
+    struct PaintStructBoundBox
+    {
+        uint16_t x;    // 0x08
+        uint16_t y;    // 0x0A
+        uint16_t z;    // 0x0C
+        uint16_t xEnd; // 0x0E
+        uint16_t yEnd; // 0x10
+        uint16_t zEnd; // 0x12
+    };
+
+    struct PaintStruct
+    {
+        uint32_t imageId; // 0x00
+        union
+        {
+            uint32_t tertiaryColour; // 0x04
+            // If masked image_id is masked_id
+            uint32_t colourImageId; // 0x04
+        };
+        PaintStructBoundBox bounds; // 0x08
+        int16_t x;                  // 0x14
+        int16_t y;                  // 0x16
+        uint16_t quadrantIndex;
+        uint8_t flags;
+        uint8_t quadrantFlags;
+        AttachedPaintStruct* attachedPS; // 0x1C
+        PaintStruct* children;
+        PaintStruct* nextQuadrantPS;                   // 0x24
+        Ui::ViewportInteraction::InteractionItem type; // 0x28
+        uint8_t var_29;
+        uint16_t pad_2A;
+        coord_t map_x; // 0x2C
+        coord_t map_y; // 0x2E
+        union
+        {
+            Map::tile_element* tileElement; // 0x30 (or thing pointer)
+            Thing* thing;                   // 0x30
+        };
+    };
+    static_assert(sizeof(PaintStruct) == 0x34);
+
+    union PaintEntry
+    {
+        PaintStruct basic;
+        AttachedPaintStruct attached;
+        PaintStringStruct string;
+    };
+    static_assert(sizeof(PaintEntry) == 0x34);
+#pragma pack(pop)
+
+    struct PaintSession
+    {
+    public:
+        void generate();
+        void arrangeStructs();
+        [[nodiscard]] Ui::ViewportInteraction::InteractionArg getNormalInteractionInfo(const uint32_t flags);
+        [[nodiscard]] Ui::ViewportInteraction::InteractionArg getStationNameInteractionInfo(const uint32_t flags);
+        [[nodiscard]] Ui::ViewportInteraction::InteractionArg getTownNameInteractionInfo(const uint32_t flags);
+
+    private:
+        //Gfx::drawpixelinfo_t* dpi;                     // 0xE0C3E0
+        //PaintEntry paintStructs[4000];                 // 0xE0C410 -> 0xE3F050
+        //PaintStruct* quadrants[1024];                  // 0xE3F0C0 -> 0xE400C0
+        //uint32_t quadrantBackIndex;                    // 0xE400C0
+        //uint32_t quadrantFrontIndex;                   // 0xE400C4
+        //const void* currentlyDrawnItem;                // 0xE4F0B4
+        //PaintEntry* endOfPaintStructArray;             // 0xE0C404
+        //PaintEntry* nextFreePaintStruct;               // 0xE0C40C
+        //Map::map_pos spritePosition;                   // 0xE3F090 and 0xE3F096
+        //PaintStruct* lastRootPS;                       // 0xE40120
+        //UI::ViewportInteraction::InteractionItem type; // 0xE3F0AC
+        //PaintStringStruct* paintStringHead;            // 0xE40118
+        //PaintStringStruct* lastPaintString;            // 0xE4011C
+        //Map::map_pos mapPosition;                      // 0xE3F0B0
```

Comment:
Could you comment on why these have been commented out?

Reply:
These are all fields that I would like the session class to own. I have a little idea for how to handle this give me a sec.